tests/Aspire.Hosting.SqlServer.Tests/SqlServerFunctionalTests.cs

```diff
@@ -0,0 +1,318 @@
+
+        await app.StartAsync(cts.Token);
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{tempDb.Resource.Name}"] = await tempDb.Resource.ConnectionStringExpression.GetValueAsync(default),
+        });
+
+        hb.AddSqlServerDbContext<TestDbContext>(tempDb.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+            await databaseCreator.CreateTablesAsync(token);
+        }, cts.Token);
+
+        var dbContext = host.Services.GetRequiredService<TestDbContext>();
+        var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+
+        dbContext.Cars.Add(new TestDbContext.Car { Brand = "BatMobile" });
+        await dbContext.SaveChangesAsync(cts.Token);
+
+        var cars = await dbContext.Cars.ToListAsync(cts.Token);
+        Assert.Single(cars);
+        Assert.Equal("BatMobile", cars[0].Brand);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifySqlServerResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var sqlserver = builder.AddSqlServer("sqlserver");
+        var tempDb = sqlserver.AddDatabase("tempdb");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{tempDb.Resource.Name}"] = await tempDb.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddSqlServerClient(tempDb.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var conn = host.Services.GetRequiredService<SqlConnection>();
+
+            if (conn.State != System.Data.ConnectionState.Open)
+            {
+                await conn.OpenAsync(token);
+            }
+
+            var selectCommand = conn.CreateCommand();
+            selectCommand.CommandText = $"SELECT 1";
+            var results = await selectCommand.ExecuteReaderAsync(token);
+
+            Assert.True(results.HasRows);
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false, Skip = "https://github.com/dotnet/aspire/issues/5055")]
```

Comment:
It seems concerning that we are changing the `WithDataBindMount` implementation in this PR, but then also shutting off the test that checks `WithDataBindMount`.

Reply:
It is concerning but it has never been enabled on the CI, and this would have to be the same for other containers: Oracle, MongoDB. I have tried many things without success, setting Unix permissions, creating the folder differently, ... Nothing works on the CI even though it works locally, both with a Windows or WSL2 test runner.

I will try on one of the perf machines which has native Linux (same 22.04 as the CI) in case I can repro the problem.